src/model.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+from market_data import MarketData
+from enums import *
+from volgrid import VolGrid
+import numpy as np
+from abc import ABC, abstractmethod
+
+class MarketModel(ABC):
+    def __init__(self, und: Stock):
+        self._und = und
+        self._interest_rate = MarketData.get_interest_rate()
+        self._initial_spot = MarketData.get_initial_spot(self._und)
+        self._volgrid = MarketData.get_volgrid(self._und)
```

Comment:
Both get_initial_spot and get_vol_grid() - note the underscore - take no arguments and returns a dictionary. Change it to:
```
self._initial_spot = MarketData.get_initial_spot()[self._und]
self._volgrid = MarketData.get_vol_grid()[self._und]
```

Reply:
changes have been implemented